% Use only LaTeX2e, calling the article.cls class and 12-point type.

\documentclass[12pt]{article}

\usepackage{scicite}
\usepackage{lineno}
\linenumbers

% Use times if you have the font installed; otherwise, comment out the
% following line.
\usepackage{times}

% useful packages
\usepackage{xcolor}
\usepackage{hyperref}
\usepackage{comment}
\usepackage{graphicx}
\usepackage{subcaption}


\topmargin 0.0cm
\oddsidemargin 0.2cm
\textwidth 16cm 
\textheight 21cm
\footskip 1.0cm


\newenvironment{sciabstract}{%
\begin{quote} \bf}
{\end{quote}}


% If your reference list includes text notes as well as references,
% include the following line; otherwise, comment it out.

\renewcommand\refname{References and Notes}

\newcommand{\adpcomment}[1]{{\textcolor{orange}{ADP: #1}}}
\newcommand{\sgcomment}[1]{\textcolor{red}{SG: #1}}

% The following lines set up an environment for the last note in the
% reference list, which commonly includes acknowledgments of funding,
% help, etc.  It's intended for users of BibTeX or the {thebibliography}
% environment.  Users who are hand-coding their references at the end
% using a list environment such as {enumerate} can simply add another
% item at the end, and it will be numbered automatically.

\newcounter{lastnote}
\newenvironment{scilastnote}{%
\setcounter{lastnote}{\value{enumiv}}%
\addtocounter{lastnote}{+1}%
\begin{list}%
{\arabic{lastnote}.}
{\setlength{\leftmargin}{.22in}}
{\setlength{\labelsep}{.5em}}}
{\end{list}}


% Include your paper's title here
\title{A review of UMAP in population genetics} 


% Place the author information here.  Please hand-code the contact
% information and notecalls; do *not* use \footnote commands.  Let the
% author contact information appear immediately below the author names
% as shown.  We would also prefer that you don't change the type-size
% settings shown here.

\author
{Alex Diaz-Papkovich,$^{1,2}$ Simon Gravel$^{2\ast}$\\
\\
\normalsize{$^{1}$Quantitative Life Sciences Program, McGill University, Montreal,}\\
\normalsize{$^{2}$Department of Human Genetics, McGill University, Montreal}\\
\\
\normalsize{$^\ast$To whom correspondence should be addressed; E-mail:  simon.gravel@mcgill.ca.}
}

% Include the date command, but leave its argument blank.

\date{}



%%%%%%%%%%%%%%%%% END OF PREAMBLE %%%%%%%%%%%%%%%%



\begin{document} 

% Double-space the manuscript.

\baselineskip24pt

% Make the title.

\maketitle 


% Place your abstract within the special {sciabstract} environment.

\begin{sciabstract}
Uniform manifold approximation and projection (UMAP) has been rapidly adopted by the population genetics community to study population structure. It has become common in visualizing the ancestral composition of human genetic datasets, as well as searching for unique clusters of data, and for identifying geographic. Here we give an overview of applications of UMAP in population genetics, provide recommendations for best practices, and offer visualizations and applications using using publicly available data.
\end{sciabstract}

\section*{Introduction}
% give brief statement on population stratification?

One of the primary challenges of genomic data analysis is high dimensionality. The human genome has over three billion base pairs, and many biobanks contain hundreds of thousands of individuals and above. Relationships among individuals are relevant for historical studies as well as for studies that seek to identify genetic roots of diseases. These relationships can be influenced by demography, by sampling strategies, as well as by technical variation. A first step in many genomic analyses is dimensional reduction approach to visualize the data to identify relevant relatedness patterns. 

One of the most common methods of dimensionality reduction is principal component analysis (PCA) to identify directions, in the high-dimensional space, along which data is most variable. The projection of genomic data along these directions provides a low-dimensional representation of the data that captures as much variance as possible. Because PCA projection is a linear operation, has a relatively straightforward interpretation in terms of demographic events \cite{mcvean2009genealogical}, and is also well-suited to the correction of population structure in genome-wide association studies (GWAS) \cite{patterson2006population}, it has become near-ubiquitous in the literature.

Dimension reduction requires tradeoffs. Because PCA projection identifies directions of maximal variance in the data and ignores other variance components, it tends to obscure finer scale patterns of variation.  Many nonlinear neighbour graph-based dimension reduction algorithms have been developed over the years to address this need. Here we focus on uniform manifold approximation and projection (UMAP)\cite{mcinnes_umap_2018}, a method developed in 2018 that has seen widespread in many fields (e.g. single-cell genomics\cite{becht_dimensionality_2019}). 

Rather than trying to preserve large-scale structure, UMAP seeks to preserve local neighbourhoods in a dataset. For each individual in a genetic dataset, UMAP identifies a pre-set number of nearest neighbours and represents distances in the local neighborhood as a weighted graph where nearest neighbours are weighted more heavily. The goal is then to find a low-dimensional representation of the data that preserves these neighbourhoods as much as possible. By focusing on preserving neighborhoods (rather than absolute distances), UMAP allows for data-dense regions to be ``stretched out'' in the representation. This has the benefit of reducing overcrowding of the low-dimensional representation, but comes at the cost of a more challenging interpretation of distances in the low-dimensional representation.  This is an important distinction relative to algorithms such as PHATE \cite{} that allow nonlinear transformations of the data while seeking to preserve distances. 
Because UMAP seeks to preserve the topology of the original dataset, rather than the absolute distances, the low dimensional representation can, in principle, contain arbitrarily small distances between points in the representation. Even though such small distances might be a faithful representation of the original data, it is not ideal for visualization. UMAP allows for specification of a minimum distance between nearest neighbours in low-dimensional space. The minimum distance threshold is useful for visualization, but can be set to zero or very small values when using the reduced dimension representation for downstream analyses (such as clustering).  



%, with values closer to zero resulting in more tightly clumped data. 
%\adpcomment{Best explanation (from the docs): The min_dist parameter controls how tightly UMAP is allowed to pack points together. It, quite literally, provides the minimum distance apart that points are allowed to be in the low dimensional representation. This means that low values of min_dist will result in clumpier embeddings. This can be useful if you are interested in clustering, or in finer topological structure. Larger values of min_dist will prevent UMAP from packing point together and will focus instead on the preservation of the broad topological structure instead.}




%draws edges connecting it to a pre-set number of its nearest neighbours and creates a weighted neighbour-graph. The weight of each edge is a value between zero and one representing whether a point belongs to a neighbourhood; this neighbourhood can be thought of as a fuzzy open set. Since these sets and their edge weights are specific to each point, they must be combined by taking their union; in this sense, UMAP patches together all neighbourhoods and creates a weighted graph for the entire dataset. The algorithm then creates a low-dimensional representation that minimizes the difference between high- and low-dimensional edge weights. By focusing on the local structure of every point, it can piece together a low-dimensional representation. 

In the context of genetic data, UMAP finds the nearest genetic neighbours for each individual and creates low-dimensional representations that group more closely-related individuals together, and partially preserves longer-range relatedness through intermediary individuals. When used in visualizations, UMAP embeddings uncover many subtle features of data, such as distinct demographic histories and covariation between genetics, geography, and phenotypes\cite{diaz-papkovich_umap_2019}. Figure~\ref{fig:PCA_and_UMAP} compares visualizations of PCA to UMAP using genotype data from the Thousand Genomes Project (1KGP)\cite{10002015global}. PCA flattens the third dimension, obscuring South Asian and Central/South American populations, whereas UMAP places them in more clearly visible clusters. UMAP has become widely used to study population structure in humans and other species, becoming a popular addition to research in conjunction with existing methods. Here we will describe the current state of the use of UMAP in population genetics.

\section*{Visualizing ancestral diversity in cohorts}
The most straightforward and common use of UMAP is for visualization. This has proven useful for data composed of relatively homogeneous populations as well as those with considerable diversity in ancestries. UMAP will dedicate more visual space to larger populations within a cohort, and consequently can illustrate the ancestral composition of a cohort in the context of its population structure as well as the size of the data. Often these data are combined with reference panels such as the 1KGP or the Human Genome Diversity Project (HGDP)\cite{cann2002human}. Just as with PCA analysis, researchers can either perform the dimensional reduction jointly or project one dataset onto UMAP embeddings of a reference data. In most surveyed literature \sgcomment{is that what you meant by "generally"?}\adpcomment{Yes}, data are restricted to common variants with a minor allele frequency (MAF) greater than some threshold, e.g. $0.01$. However, UMAP can be used on any subset of interest.

% First paragraph: first use of UMAP and what it showed
%In the first use of UMAP in population genetics, we visualized the populations in the 1KGP, Health and Retirement Study (HRS)\cite{juster1995overview}, and UK biobank (UKB)\cite{sudlow2015uk} using their genotypic data. UMAP illustrated fine-scale structure and could highlight unique distributions of phenotypes and clusters of individuals. Clusters could arise from close relations between individuals, or from unique demographic histories. We recommended to execute multiple runs of UMAP while varying parameters to judge how stable patterns were and to analyze them further with other methods\cite{diaz-papkovich_umap_2019}.

In the first use of UMAP in population genetics, we demonstrated how UMAP illustrates fine-scale structure in datasets including the 1KGP and the UK biobank(UKB)\cite{sudlow2015uk}\cite{diaz-papkovich_umap_2019}. Figures~\ref{fig:PCA_and_UMAP} and \ref{fig:UMAP_fine_scale} show how UMAP emphasizes local structure over PCA's focus on global variation, and how this generates clusters using 1KGP data. Unlike the 1KGP, the UKB data is much very ancestrally diverse and contains widely varying population sizes. Figure~\ref{fig:UKB} highlights how for such a cohort UMAP can simultaneously show the fine-scale structure within the British and Irish populations, as well as the structure of those with non-European ancestries, and the relative sizes of all the groups.

\sgcomment{I wonder if it would be worth including these large cohort plots in here, and comment on them? Right now, it is not immediately clear to me what message you are trying to get across, and you give a lot of details that might not be so useful to the reader... Maybe try to convey the big picture: When applied to large-scale, ancestrally diverse cohort, UMAP tends to reveal both continental and sub-continental clusters with rather straightforward interpretation [talk about some of the findings?}

These visualizations are now used regularly in characterizing the ancestral composition of genetic cohorts with many ancestries. The Genome Aggregation Database (gnomAD v3) from the Broad Institute visualized $141,456$ exome and whole genome sequences\cite{karczewski_mutational_2020}, and Bio\textit{Me} biobank, a multi-ethnic cohort from New York City, visualized $31,705$ genotyped individuals combined with a reference panel of $87$ individuals from $7$ continental or subcontinental regions using the 1KGP, HGDP, and the Population Architecture Using Genetics and Epidemiology (PAGE) Study\cite{matise2011next}. The Million Veterans Program (MVP) is another ethnically diverse American dataset, composed of $459,777$ veterans. About $30\%$ of the cohort is made up of individuals having African, East Asian, and Native American ancestry\cite{hunter-zinck_genotyping_2020}, making UMAP a good candidate for visualizing the array of backgrounds.

\section*{Fine-scale structure in genetics}
\sgcomment{When applied to more homogeneous groups, UMAP can reveal clusters that are readily interpretable in terms of geography (such as the Japan dataset),  .... , maybe even talk about the importance of pruning and include Luke's HLA plot (if he also wants to be an author)   }

\adpcomment{Should I define "fine-scale"? Include some context for, e.g., PRS? Not sure where to put a sentence like this.}
Fine-scale population structure continues to confound studies of polygenic traits (e.g. \cite{kerminen2019geographic}, \cite{berg2019reduced}, \cite{sohail2019polygenic}), making it an important area of study.

The projections are also useful for examining sub-contintental structure. The National Geographic Genographic Project analyzed the genotype data of $32,589$ individuals living in the United States\cite{dai_population_2020}, and to further examine subcontinental structure in those of European ancestry they jointly visualized a subet of the data with The Population Reference Sample (POPRES)\cite{nelson2008population}. Spear et al. visualized American individuals with ancestries from Central and South America and the Caribbean using data from the Hispanic Community Health Study/Study of Latinos (HCHS/SOL)\cite{sorlie2010design}. Though many individuals had overlapping global ancestry proportions, UMAP visualizations put them in separate clusters, suggesting that their demographic histories arose from different Amerindigenous groups\cite{spear2020recent} \sgcomment{Talk about follow-up analysis?}.

UMAP is effective at finding clusters in cohorts that are relatively homogeneous, such as Biobank Japan (BBJ)\cite{nagai2017overview}. Sakaue et al.\cite{sakaue_dimensionality_2020} used UMAP to identify substructure within the Japanese population, separating it into a mainland population and Hokkaido-Ainu with surrounding islands, reflecting known demographic history in Japan. When using the cluster membership as a GWAS covariate, they found significant differences in polygenic risk scores (PRS) estimated later. UMAP has been noted as a potential method to identify clusters in  genetic data when the number of clusters is not known in advance\cite{tonkin-hill_fast_2019}.

% not just for markers
Genetic data are not limited to autosomal markers. Margaryan et al\cite{margaryan_population_2019} studied the ancient genomes of $442$ individuals from across Europe and Greenland, dating from as early as c. 2400 BCE to 1600 CE. UMAP was used on the top components of a multidimensional scaling (MDS) projection, as well as an identity-by-state (IBS) matrix to help identify relationships between Viking Age Scandinavians and other contemporary European populations. 

The mitochondrial DNA (mtDNA) of $1,928$ Japanese individuals was analyzed by Yamamote et al in an effort to identify clusters within known haplogroups\cite{yamamoto_genetic_2020}. The clusters generated by UMAP were concordant with known sub-haplogroups, though it did not clearly delineate larger clusters identified with PCA and phylogenetics. They suggest using UMAP as one of several methods when studying mtDNA. Almarri et al.\cite{almarri_population_2020} studied population structure in structural variants of $991$ individuals in the Human Genome Diversity Project (HGDP-CEPH) panel and found continental separation and variants that were private to single populations. \sgcomment{It is not clear to me what UMAP brings here, and I would probably say so.}

% Nonhuman 
\sgcomment{UMAP is particularly helpful when there is a hign number of significant PCs. }
Finding fine-scale structure with UMAP can illuminate geographic relationships and shared histories between groups. This was noted in Diaz-Papkovich et al.\cite{diaz-papkovich_umap_2019} but has also been found useful in animal populations to study spatial introgression in mussels\cite{simon_local_2019}, genetic bottlenecks in the white rhino population\cite{sanchez-barreiro_historical_2020}, and the geographic origin of disease-carrying mosquitoes\cite{consortium_genome_2020}\cite{schmidt_population_2020}.

\section*{Discussion}
% Discussion structure:
% 1. Common, good practices for visualization
% 2. Common, good practices for fine-scale?
% 3. New things in UMAP (connectivity, tie into e.g. clustering)

UMAP is now regularly used to visualize the ancestral composition of cohorts as well as to examine fine-scale population structure and subtle patterns in data, as in figure~\ref{fig:PCA_and_UMAP} where the 1KGP populations form individual clusters. It is itself effective across multiple types of genetic data (e.g. genotype, WGS, mtDNA) and biobanks of all compositions. The main parameters are the number of nearest neighbours (NN) and the minimum distance (MD). Studies varied in their parameter selection, but generally chose NN close to $15$; setting $NN < 10$ can result in separated clusters made up of closely-related individuals, such as families. The minimum distance was usually $0.1 < MD < 0.5$; values of $MD$ close to $0$ create very tight clusters, while values above $0.5$ visually spread the data. This should be considered for whether UMAP will be used for visualization or a downstream process such as cluster analysis. It is good practice to run multiple parametrizations and to combine UMAP plots with PCA plots and methods like fineSTRUCTURE\cite{lawson2012inference}, ADMIXTURE\cite{alexander2009fast}, or traditional statistics such as $F_{ST}$ to make inferences.

UMAP embeddings are highly informative of population structure. While clusters will often be correlated with labels such as self-identified ethnicity or race, these are social constructs that are related to sociocultural or geographic factors that can lead to population structure, and they should not be conflated. Visualizing the simplicial complex underlying the algorithm can highlight how groups or individuals in a dataset influence clustering effects. We demonstrate this using the genotype data from the 1KGP in figure~\ref{fig:UMAP_connectivity}. Increasing the value of $NN$ increases the size of the complex, thus increasing the connectedness of clusters, though this comes at a higher computational cost. Clusters that would otherwise be separated, such as the Luhya (LWK) in figure~\ref{fig:UMAP_low_NN_1KGP}, are brought closer to other populations, shown in figure~\ref{fig:UMAP_high_NN_1KGP}. With $NN=15$, the simplicial complexes of South Asia and East Asia do not connect to other populations; that is, for each cluster, every individual's $15$ closest genetic neighbours fall within the cluster. After raising $NN$ to $200$, all continental clusters become connected, demonstrated in figures~\ref{fig:UMAP_low_NN_connectivity} and \ref{fig:UMAP_high_NN_connectivity}. \adpcomment{The EAS cluster connects to the Central/South American cluster. Worth metioning?} 

\section*{Conclusion}
With its effective performance and widespread use in under two years, UMAP shows considerable promise as part of the toolbox of a population geneticist. Beyond its capacity to visualize data, it holds promise for downstream methods such as clustering, correction for fine-scale population structure in GWAS and PRS, and identifying unique demographic histories. We anticipate that UMAP and/or related methods of dimension reduction will continue to find applications in the field, bolstering our exploration and understanding of human genomic data and the study of complex polygenic traits.

\section*{Materials and methods}
All code used to process data and generate images is available at \url{https://github.com/diazale/umap_review}. We used genotype data from $3,450$ individuals from the 1KGP using Affy 6.0 genotyping\cite{10002015global}. Genotype data from the 1KGP is available at \url{http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/supporting/hd_genotype_chip/} and \url{http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/}. Visualizations were done with matplotlib\cite{Hunter2007} and PCA was done using sklearn\cite{scikit-learn}.

%%%%% FIGURES %%%%%

\clearpage

\begin{figure}[h!]
  \centering
  \begin{subfigure}[b]{0.45\linewidth}
    \includegraphics[width=\linewidth]{code/images/1KGP_PCA.png}
    \caption{}
    \label{fig:PCA}
  \end{subfigure}
  \begin{subfigure}[b]{0.45\linewidth}
    \includegraphics[width=\linewidth]{code/images/1KGP_genotype_UMAP.png}
    \caption{}
    \label{fig:UMAP}
  \end{subfigure}
  \caption{Visualizations of data from the 1KGP. The first two principal components (left) versus a two-dimensional UMAP embedding (right).     ACB, African Caribbean in Barbados;
    ASW, African Ancestry in Southwest US;
    BEB, Bengali;
    CDX, Chinese Dai;
    CEU, Utah residents with Northern/Western European ancestry;
    CHB, Han Chinese;
    CHS, Southern Han Chinese;
    CLM, Colombian in Medellin, Colombia;
    ESN, Esan in Nigeria;
    FIN, Finnish;
    GBR, British in England and Scotland;
    GWD, Gambian;
    GTH, Gujarati;
    IBS, Iberian in Spain;
    ITU, Indian Telugu in the UK;
    JPT, Japanese;
    KHV, Kinh in Vietnam;
    LWK, Luhya in Kenya;
    MSL, Mende in Sierra Leone;
    MXL, Mexican in Los Angeles, California;
    PEL, Peruvian;
    PJL, Punjabi in Lahore, Pakistan;
    PUR, Puerto Rican;
    STU, Sri Lankan Tamil in the UK;
    TSI, Tuscani in Italy;
    YRI, Yoruba in Nigeria }
  \label{fig:PCA_and_UMAP}
\end{figure}

\clearpage

\begin{figure}
  \includegraphics[width=\linewidth]{code/images/1KGP_PCA_UMAP.png}
  \caption{UMAP on the top 15 PCs of 1KGP data results in many populations forming their own clusters. Such distinct clustering may be a consequence of the composition of the dataset, which comprises a relatively balanced sample of many different populations. Adding many of the top PCs (e.g. the top $200$) will create an embedding more similar to Fig.~\ref{fig:UMAP}. }
  \label{fig:UMAP_fine_scale}
\end{figure}

\clearpage

\begin{figure}[h!]
  \centering
  \begin{subfigure}[b]{0.49\linewidth}
    \includegraphics[width=\linewidth]{code/ukb/images/ukbb_pca_coords_eth.png}
    \caption{A PCA projection of the UKB data.}
    \label{fig:UKB_PCA}
  \end{subfigure}
  \begin{subfigure}[b]{0.49\linewidth}
    \includegraphics[width=\linewidth]{code/ukb/images/UKBB_UMAP_PC20_NC2_NN15_MD05_2018454111_eth.png}
    \caption{UMAP on the top 20 PCs of the UKB.}
    \label{fig:UKB_UMAP}
  \end{subfigure}
  \caption{PCA and UMAP projections of the UKB data, coloured by self-identified ethnic background. Unlike PCA, UMAP focuses on preserving local relationships and emphasizes fine-scale patterns in data. Groups in the UMAP projection are less compressed showing, for example, the relative size of the British and Irish populations in the UKB, alongside populations of other ancestries, while simultaneously showing the population structure between and within groups.}
  \label{fig:UKB}
\end{figure}


\clearpage

\begin{figure}[h!]
  \centering
  \begin{subfigure}[b]{0.49\linewidth}
    \includegraphics[width=\linewidth]{code/images/1KGP_genotype_UMAP_low_NN.png}
    \caption{UMAP with 15 neighbours.}
    \label{fig:UMAP_low_NN_1KGP}
  \end{subfigure}
  \begin{subfigure}[b]{0.49\linewidth}
    \includegraphics[width=\linewidth]{code/images/1KGP_genotype_UMAP_high_NN.png}
    \caption{UMAP with 200 neighbours.}
    \label{fig:UMAP_high_NN_1KGP}
  \end{subfigure}
  \begin{subfigure}[b]{0.49\linewidth}
    \includegraphics[width=\linewidth]{code/images/UMAP_connectivity_low_NN.png}
    \caption{Connectivity map of 15 neighbours.}
    \label{fig:UMAP_low_NN_connectivity}
  \end{subfigure}
  \begin{subfigure}[b]{0.49\linewidth}
    \includegraphics[width=\linewidth]{code/images/UMAP_connectivity_high_NN.png}
    \caption{Connectivity map of 200 neighbours.}
    \label{fig:UMAP_high_NN_connectivity}
  \end{subfigure}
  \caption{UMAP projection of the same genotype data from the 1KGP comparing parametrization with a small (left) and large (right) number of nearest neighbours. Top images are coloured by population; bottom images are the same points but with the simplicial complex drawn. When adding more neighbours, clusters become less separated, as with the LWK population, for example. Looking at the connectivity maps, we see new connections between continental groups, such as the Central/South American clusters and East Asian clusters. Darker lines indicate that individuals are closer to each other in genotype space.}
  \label{fig:UMAP_connectivity}
\end{figure}

\clearpage
\newpage

\bibliography{umap_popgen_review}

\bibliographystyle{Science}




\end{document}




















