% Use only LaTeX2e, calling the article.cls class and 12-point type.

\documentclass[12pt]{article}

\usepackage{scicite}
\usepackage{lineno}
\linenumbers

% Use times if you have the font installed; otherwise, comment out the
% following line.

\usepackage{times}
\usepackage{hyperref}


\topmargin 0.0cm
\oddsidemargin 0.2cm
\textwidth 16cm 
\textheight 21cm
\footskip 1.0cm


\newenvironment{sciabstract}{%
\begin{quote} \bf}
{\end{quote}}


% If your reference list includes text notes as well as references,
% include the following line; otherwise, comment it out.

\renewcommand\refname{References and Notes}

% The following lines set up an environment for the last note in the
% reference list, which commonly includes acknowledgments of funding,
% help, etc.  It's intended for users of BibTeX or the {thebibliography}
% environment.  Users who are hand-coding their references at the end
% using a list environment such as {enumerate} can simply add another
% item at the end, and it will be numbered automatically.

\newcounter{lastnote}
\newenvironment{scilastnote}{%
\setcounter{lastnote}{\value{enumiv}}%
\addtocounter{lastnote}{+1}%
\begin{list}%
{\arabic{lastnote}.}
{\setlength{\leftmargin}{.22in}}
{\setlength{\labelsep}{.5em}}}
{\end{list}}


% Include your paper's title here
\title{A review of UMAP in population genetics} 


% Place the author information here.  Please hand-code the contact
% information and notecalls; do *not* use \footnote commands.  Let the
% author contact information appear immediately below the author names
% as shown.  We would also prefer that you don't change the type-size
% settings shown here.

\author
{Alex Diaz-Papkovich,$^{1,2}$ Simon Gravel$^{2\ast}$\\
\\
\normalsize{$^{1}$Quantitative Life Sciences Program, McGill University, Montreal,}\\
\normalsize{$^{2}$Department of Human Genetics, McGill University, Montreal}\\
\\
\normalsize{$^\ast$To whom correspondence should be addressed; E-mail:  simon.gravel@mcgill.ca.}
}

% Include the date command, but leave its argument blank.

\date{}



%%%%%%%%%%%%%%%%% END OF PREAMBLE %%%%%%%%%%%%%%%%



\begin{document} 

% Double-space the manuscript.

\baselineskip24pt

% Make the title.

\maketitle 



% Place your abstract within the special {sciabstract} environment.

\begin{sciabstract}
Hello world, this is my abstract
\end{sciabstract}



% In setting up this template for *Science* papers, we've used both
% the \section* command and the \paragraph* command for topical
% divisions.  Which you use will of course depend on the type of paper
% you're writing.  Review Articles tend to have displayed headings, for
% which \section* is more appropriate; Research Articles, when they have
% formal topical divisions at all, tend to signal them with bold text
% that runs into the paragraph, for which \paragraph* is the right
% choice.  Either way, use the asterisk (*) modifier, as shown, to
% suppress numbering.

\section*{Introduction}
One of the primary challenges in genomics is its intrinsic high dimensionality. The human genome, as an example, has over three billion base pairs, and research using modern biobank data typically includes at least several hundred thousand individuals. The curse of dimensionality is that while data is plentiful, signal is sparse and often confounded. One prominent example is the confounding of GWAS results caused by population stratification, also called population structure. Thus to make effective inferences on genomic data, one must take steps to reduce its dimensionality using methods that are meaningful and practical. 

One of the most common methods of dimensionality reduction to address population structure is principal component analysis (PCA). This linear transformation projects genomic data to a set of lower dimensional planes, which are ordered such that they explain the most-to-least variance. This approach is well-suited to, for example, the correction of population structure in genome-wide association studies (GWAS).\cite{patterson2006population} Plots of genomic data transformed using PCA are near-ubiquitous in the literature.

PCA can struggle to uncover fine-scale population structure; since the mathematical approach to PCA is built on representing the global variance of data, local patterns are more difficult to parse. Many nonlinear neighbour graph-based dimension reduction algorithms have been developed over the years to address this need. Here we focus on uniform manifold approximation and projection (UMAP)\cite{mcinnes_umap_2018}, a method developed in 2018 that has seen widespread in many fields (e.g. single-cell genomics\cite{becht_dimensionality_2019}). 

UMAP models the high-dimensional topology of data by constructing a neighbourhood around each point and assigning a probability of that a pre-determined number of nearby points belong to the neighbourhood via fuzzy sets. It then optimizes a low-dimensional representation that preserves these neighbourhoods and patches them together. There are two main parameters for users to consider: the number of neighbours to include in a neighbourhood, and the minimum distance between points in the low-dimensional representation. In the context of genetic data, UMAP creates low-dimensional representations that group more closely-related individuals together while preserving longer-range relatedness through intermediary individuals. When used in visualizations, UMAP embeddings uncover many subtle features of data, such as distinct demographic histories and covariation between genetics, geography, and phenotypes\cite{diaz-papkovich_umap_2019}.

Since its first use in population genetics, UMAP has become widely used to study population structure in humans and other species. It has a become a popular addition to research in conjuction with existing methods such as Admixture, linear mixed models, and PCA itself. Here we will describe the current state of the use of UMAP in population genetics.

\section*{Overview}

Before using UMAP, researchers need to consider their input data, as this will determine the output. The studies cited here have used data from genotypes, exomes, mitochondrial DNA (mtDNA), identity-by-descent (IBD) matrices, structural variants, specific genes, and contig markers, among others. It is also common to use the top principal components of data to ensure computational feasibility, though the number of top components used varies widely. The type of data will reflect a particular aspect of population structure and may have different results when using the same samples in a study. Further, it is possible to combine ancient genomes with contemporary genomes, or to compare the genomes of different species in order to, for example, reconstruct phylogenetic trees. For clarity, this overview will be split into human and non-human studies.

\subsection*{Human genetics}
\subsubsection*{Visualizing ancestral diversity in cohorts}
The most straightforward and common use of UMAP is for visualization, and with human data it is often used to illustrate and characterize the ancestral composition of a cohort. This has proven useful both for data composed of relatively homogeneous populations as well as those with considerable diversity in ancestries. Often these data are used with reference panels such as the 1000 Genomes Project (1KGP)\cite{10002015global} and Human Genome Diversity Project (HGDP)\cite{cann2002human}. Researchers may use them jointly or project their data onto UMAP embeddings of reference data, or vice versa. Generally, data is restricted to common variants with a minor allele frequency (MAF) greater than some threshold, e.g. $0.01$.

(Insert 1KGP figure here?)

The first use of UMAP in population genetics was in 2019 by testing the method on the 1KGP, Health and Retirement Study (HRS)\cite{juster1995overview}, and the UK biobank (UKB)\cite{sudlow2015uk}. It was found that UMAP, especially in combination with PCA, was valuable in visualizing diverse cohorts and could be used to highlight unique distributions of phenotypes, clusters of individuals, and gradients of admixture. The authors recommended using multiple runs with varying parameterizations and combining visualizations with other forms of analysis to derive scientifically sound conclusions\cite{diaz-papkovich_umap_2019}..

The third version of Genome Aggregation Database (gnomAD v3) from the Broad Institute contains data on $125,748$ exomes and $15,708$ whole genomes and used UMAP on the top ten principal components to illustrate the ancestral diversity of the cohort\cite{karczewski_mutational_2020}. Similarly, the Bio\textit{Me} biobank is a multi-ethnic cohort curated by the Mount Sinai Health System in New York City with $36,061$ individuals linked to electronic health records. In order to visualize the cohort's fine-scale structure, they used UMAP on the top ten principal components of $31,705$ genotyped individuals combined with a reference panel of $87$ individuals from $7$ continental or subcontinental regions using the 1KGP, HGDP, and the Population Architecture Using Genetics and Epidemiology (PAGE) Study\cite{matise2011next}. They filtered the data to remove regions of recent selection and regions of extended linkage disequilibrium resulting in $260,502$ SNPs.
%Belbin et al\cite{belbin_towards_2019} linked $36,061$ individuals in the Bio\emph{Me} dataset to New York City's Mount Sinai Health System and compared self-reported ethnicity in Bio\emph{Me} enrollment to the ethnicity in the health system's electronic health records. In characterizing the genetic ancestry of individuals, they used $31,705$ genotyped individuals combined with a reference panel of $87$ individuals from $7$ continental or subcontinental regions.
%\begin{itemize}
%\item{$MAF > 0.01$}
%\item{Removed regions of recent selection (HLA, LCT, chr8 inv, extended LD on chr17, EDAR, SLC2A5, TRBV9}
%\item{Intersected with $26$ populations from 1KGP(3), 53 from HGDP, 8 from PAGE}
%\item{n=260502 SNPs and N=35854 individuals}
%\item{UMAP on top 10 PCs, used default settings in R}
%\end{itemize}

The National Geographic Genographic Project analyzed the genotype data of $32,589$ individuals living in the United States\cite{dai_population_2020}. To examine continental structure, they jointly used the top 20 PCs from their data as well as the 1KGP, and further probed subcontinental structure in European individuals by jointly visualizing the data with The Population Reference Sample (POPRES)\cite{nelson2008population}. These were combined with ADMIXTURE, fineSTRUCTURE, and traditional PCA to infer ancestries and visualize the cohort's diversity.

The Million Veterans Program (MVP) is another ethnically diverse American dataset, composed of $459,777$ veterans and $668,418$ genetic markers. The cohort is approximately $71\%$ individuals of European ancestry with the remaining individuals having significant African, East Asian, and Native American ancestry. They categorized ancestries based on PCA and global proportions calculated in ADMIXTURE using reference populations from the 1KGP and created a UMAP embedding using the top 10 PCs.\cite{hunter-zinck_genotyping_2020}

Americans with ancestry from Central America, South America, and the Caribbean are commonly grouped as ``Hispanic'' or ``Latino'', but their genetic ancestries vary widely. One study of these populations visualized the ancestral diversity in the Hispanic Community Health Study/Study of Latinos (HCHS/SOL) \cite{sorlie2010design}. Using UMAP on the top three PCs, the authors found that while some groups seemed similar when examining proportions of global ancestry, using UMAP put them in separate clusters, suggesting that their demographic histories were from different Amerindigenous groups.\cite{spear2020recent}

\*subsubsection{Population structure of types of genetic data}

In the Almarri study, they took the Human Genome Diversity Project (HGDP-CEPH) panel of 911 samples from 54 populations worldwide and examined population structure within the structural variants. A UMAP projection of deletion variants in genotype data found separation on the continental and even individual population level variants. Other structural variants, such as insertions, duplications, multiallelic variants, and inversions also show population structure, though to a lesser degree than deletions. This allowed the investigation of variants that were private to specific populations. UMAP provided an advantage as a high number of principal components showed structure. The number of PCs selected varied by the type of structural variant, ranging from 4 biallelic duplications to 20 for deletion genotypes.\cite{almarri_population_2020}

The Japanese population is often used as 

mtDNA of the Japanese population has been studied. Identified sub-haplogroups. Used it to cluster into groups (identified 20+) that were concordant with existing haplogroups. THey recommend using a variety of dimension reduction and clustering algorithms to better understand clustering and structure in mtDNA data.\cite{yamamoto_genetic_2020} Test a variety of dimension reduction methods on BBJ data. Motivated to study population structure as it affects biomedical studies. Found substructure within Japanese population. Separated Japanese population into mainland and Hokkaido-Ainu plus surrounding islands. Reflected the demographic history of Japan. When using the cluster membership as a GWAS covariate, they found significant differences in PRS.\cite{sakaue_dimensionality_2020}

Margaryan et al\cite{margaryan_population_2019} studied the ancient genomes of $442$ individuals from across Europe and Greenland, dating from as early as c. 2400 BCE to 1600 CE. UMAP was used as part of a series of methods to identify relationships between Viking Age Scandinavians and other contemporary European populations. Running UMAP on the top $10$ components of an MDS projection of IBS data, they found clusters. Also ran UMAP on IBS data. Images were followed up with analysis. Supplemented with $f_4$ statistics, ChromoPainter, fineStructure, admixture analysis.

\subsection*{Non-human genetics}

Study of marine mussel species and spatial introgression. Took the top 11 PCs and based this on the expectation that there would be 12 panmictic populations. Correctly note that UMAP doesn't conserve distances but groups local neighbours. To filter out linkage, PCA was carried out on markers of different physical contigs, retaining those with the least missing data. The final data were 160 markers. Further analysis was carried out with Structure. \cite{simon_local_2019}

Unsupervised machine learning has been broached as a method for species delineation. In one study, it was used on a taxon of arachnids with significant genetic structure (Opiliones, Laniatores, Metanonychus). Note: this one used more t-SNE than UMAP, but noted that UMAP was a new method that could be used.\cite{derkarabetian_demonstration_2019}

Taxonomic work. Briefly mentions UMAP as one of many approaches to visualize population structure. \cite{greenbaum_network-based_2019}

Motivated to develop a method of clustering for multi-locus genotype data. The data they used here was $110,000$ sequences of HIV-1 Pol genes as well as several bacterial datasets. UMAP was noted as being useful for when model-based methods do not provide a clear answer as to how many population clusters one should expect. Probably relevant for population genetics as selecting $K$ is arbitrary. Also the visualizations are good for variations within populations.\cite{tonkin-hill_fast_2019}

% unclear what they used this for -- referencing non-existant supp figure?
%Constructed artificial genomes and used UMAP to measure whether they were reconstructing population structure accurately.\cite{yelmen_creating_2019}

The The Anopheles gambiae 1000 Genomes (Ag1000G) project aims to build a large scale biobank of mosquito genomes. In phase 2 of the project, they analyzed the whole genomes of $1,142$ wild and $234$ laboratory mosquitoes of \textit{Anopheles gambiae} and \textit{Anopheles coluzzii}, the major African malaria vector species. UMAP was carried out on biallelic SNPs of the euchromatic regions of Chromosome 3 in order to visualize population structure among the species and analyse whether there were differences among mosquitoes between countries. The data were pruned for LD. The dimension reduction was supplemented with PCA and Admixture analysis as well to find/corroborate population structure. Also used AIM and $F_{ST}$ and downsampled so that population sizes would be equal and take up the same space.\cite{consortium_genome_2020}


UMAP was also used to study the invasive arbovirus vectors \textit{Aedes aegypti} and \textit{Ae. albopictus}. These mosquitoes can spread through human travel and the Indo-Pacific is responsible for $70\%$ of global dengue transmission. Four hundred eighty mosquitoes were sampled from the Indo-Pacific to study their genetic differences. The PCA-UMAP approach was supplemented with discriminant analysis of principal components (DAPC[57]), and sparse non-negative matrix factorisation (sNMF [58]). They used the top 5 PCs. The UMAP projections were used to track genetic relatedness between invasive species.\cite{schmidt_population_2020}

The northern and southern white rhinoceros are related to each other but geographically isolated with no mating within the last $10,000$ years. The northern species has been declared extinct in the wild, while the souther species has survived after a severe population bottleneck. In addition to PCA and Admixture analysis, UMAP on modern and historical samples from museums revealed geographic substructure that predated the extinction of the north species and the bottleneck of the souther species.\cite{sanchez-barreiro_historical_2020}

\section*{Discussion}

-Note: Need to have a comment on UMAP/clustering and race

-UMAP works best in conjunction with other methods to follow up on signals.

-Combinations of data sources (reference panels, cohorts, ancient and modern genomes)

-Make some notes on how input->output. Specifically that varying parameters and input data will result in different outputs (e.g. different clusters). A lot of data gets compressed down to two dimensions. Similar to how maps get distorted from 3D to 2D, we must necessarily distort the relationships between points (individuals and groups).

%\section*{Formatting Citations}

%You can also generate your reference lists by using the list environment \texttt{\{thebibliography\}} at the end of your source document; here again, you may find the \texttt{scicite.sty} file useful.

\bibliography{umap_popgen_review}

\bibliographystyle{Science}



% Following is a new environment, {scilastnote}, that's defined in the
% preamble and that allows authors to add a reference at the end of the
% list that's not signaled in the text; such references are used in
% *Science* for acknowledgments of funding, help, etc.

\begin{scilastnote}
\item We've included in the template file \texttt{scifile.tex} a new
environment, \texttt{\{scilastnote\}}, that generates a numbered final
citation without a corresponding signal in the text.  This environment
can be used to generate a final numbered reference containing
acknowledgments, sources of funding, and the like, per {\it Science\/}
style.
\end{scilastnote}




% For your review copy (i.e., the file you initially send in for
% evaluation), you can use the {figure} environment and the
% \includegraphics command to stream your figures into the text, placing
% all figures at the end.  For the final, revised manuscript for
% acceptance and production, however, PostScript or other graphics
% should not be streamed into your compliled file.  Instead, set
% captions as simple paragraphs (with a \noindent tag), setting them
% off from the rest of the text with a \clearpage as shown  below, and
% submit figures as separate files according to the Art Department's
% instructions.


\clearpage

\noindent {\bf Fig. 1.} Please do not use figure environments to set
up your figures in the final (post-peer-review) draft, do not include graphics in your
source code, and do not cite figures in the text using \LaTeX\
\verb+\ref+ commands.  Instead, simply refer to the figure numbers in
the text per {\it Science\/} style, and include the list of captions at
the end of the document, coded as ordinary paragraphs as shown in the
\texttt{scifile.tex} template file.  Your actual figure files should
be submitted separately.



\end{document}




















